src/crc64.c

```diff
@@ -126,33 +143,225 @@ uint64_t crc64(uint64_t crc, const unsigned char *s, uint64_t l) {
-    printf("[calcula]: c7794709e69683b3 == %016" PRIx64 "\n",
-           (uint64_t)_crc64(0, li, sizeof(li)));
-    printf("[64speed]: c7794709e69683b3 == %016" PRIx64 "\n",
-           (uint64_t)crc64(0, (unsigned char*)li, sizeof(li)));
+
+    uint64_t crc64_test_size = 0;
+    int i, lastarg, csv = 0, loop = 0, combine = 0;
+again:
+    for (i = 1; i < argc; i++) {
+        lastarg = (i == (argc-1));
+        if (!strcmp(argv[i],"--help")) {
+            goto usage;
+        } else if (!strcmp(argv[i],"--csv")) {
+            csv = 1;
+        } else if (!strcmp(argv[i],"-l")) {
+            loop = 1;
+        } else if (!strcmp(argv[i],"--crc")) {
+            if (lastarg) goto invalid;
+            crc64_test_size = atoll(argv[++i]);
+        } else if (!strcmp(argv[i],"--combine")) {
+            combine = 1;
+        } else if (!strcmp(argv[i],"test") || !strcmp(argv[i],"crc64")) {
+            // used to call this stuff from `redis-server test crc64`
+        } else {
+invalid:
+            printf("Invalid option \"%s\" or option argument missing\n\n",argv[i]);
+usage:
+            printf(
+"Usage: crc64 [OPTIONS]\n\n"
+" --csv              Output in CSV format\n"
+" -l                 Loop. Run the tests forever\n"
+" --crc <bytes>      Benchmark crc64 faster options, using a buffer this big, and quit when done.\n"
+" --combine          Benchmark crc64 combine value ranges and timings.\n"
+            );
+            return 1;
+        }
+    }
+
+    if (crc64_test_size == 0 && combine == 0) {
+        crc64_init();
+        printf("[calcula]: e9c6d914c4b8d9ca == %016" PRIx64 "\n",
+            (uint64_t)_crc64(0, "123456789", 9));
+        printf("[64speed]: e9c6d914c4b8d9ca == %016" PRIx64 "\n",
+            (uint64_t)crc64(0, (unsigned char*)"123456789", 9));
+        char li[] = "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed "
+                    "do eiusmod tempor incididunt ut labore et dolore magna "
+                    "aliqua. Ut enim ad minim veniam, quis nostrud exercitation "
+                    "ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis "
+                    "aute irure dolor in reprehenderit in voluptate velit esse "
+                    "cillum dolore eu fugiat nulla pariatur. Excepteur sint "
+                    "occaecat cupidatat non proident, sunt in culpa qui officia "
+                    "deserunt mollit anim id est laborum.";
+        printf("[calcula]: c7794709e69683b3 == %016" PRIx64 "\n",
+            (uint64_t)_crc64(0, li, sizeof(li)));
+        printf("[64speed]: c7794709e69683b3 == %016" PRIx64 "\n",
+            (uint64_t)crc64(0, (unsigned char*)li, sizeof(li)));
+        return 0;
+
+    }
+
+    int init_this_loop = 1;
+    long long init_start, init_end, mismatched = 0;
+
+    do {
+        unsigned char* data = NULL;
+        uint64_t passes = 0;
+        if (crc64_test_size) {
+            data = zmalloc(crc64_test_size);
+            if (!data) {
```

Comment:
```suggestion
            data = zmalloc(crc64_test_size);
            if (!data) {
```
zmalloc can not turn NULL, it will hit the OOM handler first. i assume it's fine to just let it crash, otherwise you can use ztrymalloc.